spacy_curated_transformers/cli/fill_config_transformer.py

```diff
@@ -0,0 +1,438 @@
+        config_overrides=overrides,
+    )
+
+
+CURATED_TRANSFORMER_TO_HF_MODEL_TYPE: Dict[str, str] = {
+    "AlbertTransformer": "albert",
+    "BertTransformer": "bert",
+    "CamembertTransformer": "camembert",
+    "RobertaTransformer": "roberta",
+    "XlmrTransformer": "xlm-roberta",
+}
+
+
+class HfParamSource(Enum):
+    MODEL_CONFIG = 1
+    TOKENIZER_CONFIG = 2
+
+
+# Entrypoint parameters that are common to all curated transformer models.
+COMMON_ENTRYPOINT_PARAMS: Dict[str, HfParamSource] = {
+    "attention_probs_dropout_prob": HfParamSource.MODEL_CONFIG,
+    "hidden_act": HfParamSource.MODEL_CONFIG,
+    "hidden_dropout_prob": HfParamSource.MODEL_CONFIG,
+    "hidden_width": HfParamSource.MODEL_CONFIG,
+    "intermediate_width": HfParamSource.MODEL_CONFIG,
+    "layer_norm_eps": HfParamSource.MODEL_CONFIG,
+    "max_position_embeddings": HfParamSource.MODEL_CONFIG,
+    "model_max_length": HfParamSource.MODEL_CONFIG,
+    "num_attention_heads": HfParamSource.MODEL_CONFIG,
+    "num_hidden_layers": HfParamSource.MODEL_CONFIG,
+    "padding_idx": HfParamSource.MODEL_CONFIG,
+    "type_vocab_size": HfParamSource.MODEL_CONFIG,
+    "vocab_size": HfParamSource.MODEL_CONFIG,
+    "model_max_length": HfParamSource.TOKENIZER_CONFIG,
+}
+
+MODEL_SPECIFIC_ENTRYPOINT_PARAMS: Dict[str, Dict[str, HfParamSource]] = {
+    "albert": {
+        "embedding_width": HfParamSource.MODEL_CONFIG,
+        "num_hidden_groups": HfParamSource.MODEL_CONFIG,
+    }
+}
+
+ENTRYPOINT_PARAMS_TO_HF_CONFIG_KEYS: Dict[str, str] = {
+    "hidden_width": "hidden_size",
+    "intermediate_width": "intermediate_size",
+    "padding_idx": "pad_token_id",
+    "embedding_width": "embedding_size",
+}
+
+
+class ModelSource(Enum):
+    CliArgument = 1
+    Loader = 2
+
+
+def init_fill_config_transformer(
+    config_path: Path,
+    output_path: Path,
+    cli_model_name: Optional[str],
+    cli_model_revision: Optional[str],
+    cli_transformer_name: Optional[str],
+    config_overrides: Dict[str, Any],
+):
+    msg = Printer()
+    _validate_hf_packages(msg)
+
+    config = util.load_config(config_path, overrides=config_overrides)
+
+    transformer_name = _resolve_curated_trf_pipe_name(msg, config, cli_transformer_name)
+    model_src = _resolve_model_source(msg, cli_model_name, cli_model_revision)
+    model_name, model_revision = _resolve_model_name_and_revision(
+        msg, config, model_src, cli_model_name, cli_model_revision, transformer_name
+    )
+
+    hf_model_type = _lookup_hf_model_type_for_curated_architecture(
+        msg, config, transformer_name
+    )
+    hf_config = _get_hf_model_config(model_name, model_revision)
+    incoming_hf_model_type = hf_config["model_type"]
+    if incoming_hf_model_type != hf_model_type:
+        msg.fail(
+            f"Hugging Face model of type '{incoming_hf_model_type}' cannot be loaded into "
+            f"Curated Transformer pipe '{transformer_name}' due to mismatching architectures",
```

Comment:
Give more pointers about what to do here, too, at the very least point to the explicit section of the config they need to look at. Probably better something like a link to the docs, since it may be too much for this message otherwise?

Reply:
I've expanded the error a bit to mention which config value they should look at. However, I didn't add a link as the docs PR is still in the works (hyperlinks/ids will most likely change before it's merged).

Reply:
Is there a way to identify the possible list of architectures from the registry and provide those?

Reply:
It might be possible, but I'm not sure if that alone would be informative enough to the user?